src/BrowserFetcher.ts

```diff
@@ -321,10 +321,43 @@ function install(archivePath: string, folderPath: string): Promise<unknown> {
 }
 
 async function extractZip(zipPath: string, folderPath: string): Promise<void> {
-  try {
-    await extract(zipPath, {dir: folderPath});
-  } catch (error) {
-    return error;
+  const nodeVersion = process.version;
+
+  /* There is currently a bug with extract-zip and Node v14.0.0 that
+   * causes extractZip to silently fail:
+   * https://github.com/puppeteer/puppeteer/issues/5719
+   *
+   * Rather than silenty fail if the user is on Node 14 we instead
+   * detect that and throw an error directing the user to that bug. The
+   * rejection message below is surfaced to the user in the command
+   * line.
+   *
+   * The issue seems to be in streams never resolving so we wrap the
+   * call in a timeout and give it 100ms to resolve before deciding on
+   * an error.
+   *
+   * If the user is on Node < 14 we maintain the behaviour we had before
+   * this patch.
+   */
+  if (nodeVersion.startsWith('v14.')) {
+    await new Promise((resolve, reject) => {
+      let extractedResolved = false;
+      setTimeout(async() => {
+        await extract(zipPath, {dir: folderPath});
+        extractedResolved = true;
+      }, 10);
+
+      if (extractedResolved)
+        resolve();
+      else
+        reject(`Puppeteer currently does not work on Node v14 due to an upstream bug. Please see: https://github.com/puppeteer/puppeteer/issues/5719 for details.`);
+    });
+  } else {
+    try {
+      await extract(zipPath, {dir: folderPath});
+    } catch (error) {
+      return error;
```

Comment:
Why not throw the error here but return it?
I think this will mask error from `extract`

Reply:
Yup, this is a problem, but this code path is the codepath we already ship and changing this unfortunately doesn't solve the problem we're seeing (the problem is that extract-zip is failing silently).

I have this on my list to fix as you're right, it silences errors, but this code has been like this for a long time so I'll leave it for now.